### PR TITLE
Allow returning result for write queries

### PR DESCRIPTION
**Pull request by MonilBhavsar**
### Details
SQLite supports RETURNING clause(https://www.sqlite.org/lang_returning.html) that supports both read and write operations at a time.
Modify the SQLite handler to allow passing result reference for WRITE queries

### Fixed Issues
Required for  https://github.com/Expensify/App/issues/51832

### Tests

Tests with PR https://github.com/Expensify/Auth/pull/13393
_________
**Internal Testing Reminder:** when changing bedrock, please compile auth against your new changes

### CODE REVIEW

**Comment by MonilBhavsar** [comment]
@tylerkaraszewski will you please review this

File: sqlitecluster/SQLite.cpp [hunk 1]
```diff
@@ -566,18 +566,25 @@ bool SQLite::write(const string& query) {
     }
 
     // This is literally identical to the idempotent version except for the check for _noopUpdateMode.
-    return _writeIdempotent(query);
+    SQResult ignore;
+    return _writeIdempotent(query, ignore);
```
Comment (tylerkaraszewski) [inline sqlitecluster/SQLite.cpp:570]: Why do we not support the new parameter for `write`?

**Review by tylerkaraszewski — CHANGES_REQUESTED** [review]
Please add tests.

File: sqlitecluster/SQLite.cpp (hunk 1 above)
Comment (MonilBhavsar) [reply sqlitecluster/SQLite.cpp:570]: We could, but I didn't see any requirement for it to make this auth flow work https://github.com/Expensify/Auth/pull/13393
May be I am missing something?

Comment (tylerkaraszewski) [reply sqlitecluster/SQLite.cpp:570]: seems weird to have it in one API call but not the other. It's not a bug, it's just inconsistent.

**Review by tylerkaraszewski — APPROVED** [review]
I've approved this

File: sqlitecluster/SQLite.cpp (hunk 1 above)
Comment (MonilBhavsar) [reply sqlitecluster/SQLite.cpp]: Got it, updated `write()`

**Pull request merged**